src/lib/components/workflow/workflows-summary-configurable-table/workflows-summary-configurable-table.svelte

```diff
@@ -0,0 +1,468 @@
+      {#each $availableWorkflowColumns as { label }, index}
+        <OrderableListItem static {index} on:addItem={() => addColumn(label)}>
+          {label}
+        </OrderableListItem>
+      {:else}
+        <OrderableListItem readonly
+          >All available headings are in view</OrderableListItem
+        >
+      {/each}
+    </OrderableList>
+    <OrderableList>
+      <svelte:fragment slot="heading">
+        Custom Search Attributes <span class="font-normal">(not in view)</span>
+      </svelte:fragment>
+      {#each $availableSearchAttributes as searchAttribute, index}
+        <OrderableListItem
+          static
+          {index}
+          on:addItem={() => addColumn(searchAttribute)}
+        >
+          {searchAttribute}
+        </OrderableListItem>
+      {:else}
+        <OrderableListItem readonly
+          >No Custom Search Attributes</OrderableListItem
+        >
+      {/each}
+    </OrderableList>
+  </div>
+</Drawer>
+
+<style lang="postcss">
+  .workflow-summary-tables-wrapper {
+    @apply relative flex flex-row w-full rounded-lg border-primary border-2 overflow-scroll;
+  }
+
+  .workflow-summary-table-wrapper {
+    @apply relative flex bg-white;
+
+    &.pinned {
+      @apply rounded-l-lg;
+
+      &::after {
+        @apply absolute right-0 content-[''] bg-primary w-[3px] h-full;
+      }
+
+      &.batch-actions-visible {
+        @apply !w-full;
+
+        &::after {
+          @apply pointer-events-none;
+        }
+      }
+
+      /* when the user has no pinned columns, hard code the wrapper to the width of the checkbox column, or 0 */
+      &.no-columns-pinned {
+        &.batch-actions-enabled {
+          @apply !w-[40px];
+        }
+
+        &:not(.batch-actions-enabled) {
+          @apply hidden;
+        }
+
+        &::after {
+          @apply pointer-events-none;
+        }
+      }
+    }
+
+    &:not(.pinned) {
+      @apply overflow-x-scroll overscroll-contain flex-grow rounded-r;
+    }
+  }
+
+  .workflow-summary-table {
+    &.pinned {
+      @apply rounded-l-lg;
+    }
+
+    &:not(.pinned) {
+      @apply rounded-r-lg table-auto w-full;
+    }
+  }
+
+  .workflow-summary-header-row {
+    @apply bg-primary text-white h-10;
+
+    &.pinned {
+      &.batch-actions-visible {
+        @apply relative z-10;
+      }
+
+      :global(th) {
+        @apply first:rounded-tl;
+      }
+    }
+
+    &:not(.pinned) {
+      :global(th) {
+        @apply last:rounded-tr;
+      }
+    }
+  }
+
+  .workflow-summary-configurable-row {
+    @apply border-b border-primary cursor-pointer h-10;
+
+    &:last-of-type {
+      @apply border-b-0;
+
+      &.pinned {
+        :global(td) {
+          @apply first:rounded-bl-lg;
+        }
+      }
+
+      &:not(.pinned) {
+        :global(td) {
+          @apply last:rounded-br-lg;
+        }
+      }
+    }
+  }
+
+  .workflow-summary-configurable-row:hover {
+    @apply bg-gradient-to-br from-blue-100 to-purple-100 bg-fixed;
```

Comment:
It might be nice to have hover for the whole row (both pinned section and the rest of the table), but it might not be worth the hassle 😅

Reply:
Hah yeah I noticed this as well, but I'm not sure it's possible with css alone and could involve some hacky JS considering we would have to apply the bg gradient to rows in 2 separate tables.

Reply:
That said, I almost prefer the separate table row hover states cause it's a further indication that one section is "pinned" while the other can be scrolled 🤷